File: mobile/android/focus-android/service-telemetry/src/main/java/org/mozilla/telemetry/util/StringUtils.java
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

package org.mozilla.telemetry.util;

import androidx.annotation.NonNull;
import androidx.annotation.RestrictTo;

@RestrictTo(RestrictTo.Scope.LIBRARY)
public class StringUtils {
    public static String safeSubstring(@NonNull final String str, final int start, final int end) {
        return str.substring(
                Math.max(0, start),
                Math.min(end, str.length()));
    }
}

File: mobile/android/android-components/components/service/glean/src/main/java/mozilla/components/service/glean/utils/StringUtils.kt
/* This Source Code Form is subject to the terms of the Mozilla Public
* License, v. 2.0. If a copy of the MPL was not distributed with this
* file, You can obtain one at http://mozilla.org/MPL/2.0/. */

package mozilla.components.service.glean.utils

object StringUtils {
    fun safeSubstring(str: String, start: Int, end: Int): String {
        return str.substring(
            Math.max(0, start),
            Math.min(end, str.length)
        )
    }
}
